### PR TITLE
`<Notification/>` - changed timeout behavior with `upgrade` prop 

### PR DESCRIPTION
**Pull request by dorduch**
### What changed

- Added `upgrade` prop due to breaking changes
- When `upgrade=true` and not providing `timeout` prop, the notification will stay persistent until closed.

- [ ] Change is tested
- [ ] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

**Comment by hepiyellow** [comment]
So...
Old behaviour:

| type | timeout provided | result|
|----|----|------------|
|global | - | no close |
|global | XX | XX timeout |
|!global | - | default 6sec timeout |
|!global | XX | XX timeout |

Old behaviour: (with upgrade=true):

| type | timeout provided| closeTimer|
|----|----|------------|
|global | - | no close |
|global | XX | XX timeout |
|!global | - | no close |
|!global | XX | XX timeout |

@wuwa @milkyfruit @LittleTal Do you confirm?

**Comment by milkyfruit** [comment]
@hepiyellow @dorduch looks right.

**Comment by hepiyellow** [comment]
@milkyfruit `upgrade` is a common prop we use when we are making breaking changes. So that people can "opt-in" to a new feature before upgrading to the next major version.

@dorduch We are preparing the next major version (version 6) nowadays. 
Can you switch the target branch to `version_6` and remove the `upgrade` prop, just change the behaviour.

File: src/Notification/Notification.spec.js [hunk 1]
```diff
@@ -222,104 +222,117 @@ describe('Notification', () => {
     });
 
     ['local', 'sticky', 'global'].forEach(type => {
-      describe(`Closing after timeout for ${type} Notification`, () => {
-        const someTimeout = 132;
+      describe(
```
Comment (hepiyellow) [inline src/Notification/Notification.spec.js]: ## Motivation

We want to make it "very-easy!" to remove deprecated code and deprecated tests later on.
So...


## How?

This describes block for all type which checks for `upgrade` in certain tests is too complicated.
- [ ] Take original describe block and copy it. Rename it to something like "deprecated closing logic".
- [ ] Write a new test which only tests the component **with** the `upgrade` prop.

> Sometimes if a lot had changed, we may extract (copy) ALL the test suit into a `Component.deprecated.spec.js`.

File: src/Notification/README.MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+### Steps to upgrade
+
+- Pass `upgrade=true` (`<Notification upgrade/>`)
+- If you didn't passed `timeout` prop and expected the notification to be hidden after 6 seconds, you should pass `timeout=Notification.DEFAULT_TIMEOUT`.
+
+> The new API will be affective from the next major version.
+
+### Documentation
+
```
Comment (hepiyellow) [inline src/Notification/README.MIGRATION.md]: Maybe add this Table of the previous logic

| type | timeout provided | closeTimer|
|----|----|------------|
|global | - | no close |
|global | XX | XX timeout |
|local , sticky | - | default 6sec timeout |
|local , sticky | XX | XX timeout |

**Comment by dorduch** [comment]
@hepiyellow  regarding the tests refactor - I just copied the existing tests and changed the test that is now in line 372, the rest is exactly as it was (with only `upgrade = true` addition) . 
Do you want me to refactor this entire `describe` block?

File: src/Notification/Notification.spec.js [hunk 3]
```diff
@@ -222,104 +222,240 @@ describe('Notification', () => {
...
+        expect(driver.visible()).toBeFalsy();
+      });
+
+      it('should allow reopening the notification after closed by close button', () => {
+        rerender(renderNotificationWithProps({ show: true, upgrade: true }));
+        expect(driver.visible()).toBeTruthy();
+      });
+    });
+
+    ['local', 'sticky', 'global'].forEach(type => {
```
[the remark this answers is not in the document]
Comment (dorduch) [reply src/Notification/Notification.spec.js]: don't we want to test each type for regression?

**Comment by hepiyellow** [comment]
@dorduch as you wish. If you dont have time then i’ll do it.

**Comment by hepiyellow** [comment]
@dorduch Will you continue this? Or should I?

**Comment by dorduch** [comment]
@hepiyellow I'll continue this.

File: src/Notification/Notification.spec.js [hunk 4]
```diff
@@ -222,104 +222,240 @@ describe('Notification', () => {
...
+          driver: notificationDriver,
+          rerender: notificationRerender,
+        } = render(renderNotificationWithProps({ show: true, upgrade: true }));
+        notificationDriver.clickOnCloseButton();
+        driver = notificationDriver;
+        rerender = notificationRerender;
+      });
+
+      beforeEach(() => {
+        jest.runAllTimers();
```
[the remark this answers is not in the document]
Comment (hepiyellow) [reply src/Notification/Notification.spec.js]: I fixed the 2nd test here

**Comment by hepiyellow** [comment]
@dorduch Can you review my changes?

**Pull request merged**